src/MGroup.FEM.ConvectionDiffusion/Isoparametric/ConvectionDiffusionElement3D.cs

```diff
@@ -0,0 +1,285 @@
+
+			for (int gp = 0; gp < QuadratureForConsistentMass.IntegrationPoints.Count; ++gp)
+			{
+				Matrix shapeFunctionMatrix = BuildShapeFunctionMatrix(shapeFunctions[gp]);
+				Matrix partial = shapeFunctionMatrix.Transpose() * shapeFunctionMatrix;
+				var jacobian = new IsoparametricJacobian3D(Nodes, shapeGradientsNatural[gp], 1e-20);
+				double dA = jacobian.DirectDeterminant * QuadratureForConsistentMass.IntegrationPoints[gp].Weight;
+				production.AxpyIntoThis(partial, dA);
+			}
+
+			production.ScaleIntoThis(material.DependentSourceCoeff * (-1d));
+			return production;
+		}
+
+		public double[] BuildProductionVector()
+		{
+			int numDofs = Nodes.Count;
+			var productionVector = Matrix.CreateZero(numDofs, 1);
+			var identity = Matrix.CreateIdentity(numDofs);
+			IReadOnlyList<double[]> shapeFunctions =
+				Interpolation.EvaluateFunctionsAtGaussPoints(QuadratureForConsistentMass);
+			IReadOnlyList<Matrix> shapeGradientsNatural =
+				Interpolation.EvaluateNaturalGradientsAtGaussPoints(QuadratureForConsistentMass);
+
+			for (int gp = 0; gp < QuadratureForConsistentMass.IntegrationPoints.Count; ++gp)
+			{
+				Matrix shapeFunctionMatrix = BuildShapeFunctionMatrix(shapeFunctions[gp]).Transpose();
+				var jacobian = new IsoparametricJacobian3D(Nodes, shapeGradientsNatural[gp], 1e-20);
+				double dA = jacobian.DirectDeterminant * QuadratureForConsistentMass.IntegrationPoints[gp].Weight;
+				productionVector.AxpyIntoThis(shapeFunctionMatrix, dA);
+			}
+
+			productionVector.ScaleIntoThis(material.IndependentSourceCoeff);
+
+			double[,] vectorDouble = productionVector.CopyToArray2D();
+			double[] result = new double[numDofs];
+
+			for (int i = 0; i < numDofs; i++)
+			{
+				result[i] = vectorDouble[i, 0];
+			}
+			return result;
+		}
+
+		private Matrix BuildDeformationMatrix(Matrix shapeGradientsCartesian)
+		{
+			var deformation = Matrix.CreateZero(3, Nodes.Count);
+			for (int nodeIdx = 0; nodeIdx < Nodes.Count; ++nodeIdx)
+			{
+				deformation[0, nodeIdx] = shapeGradientsCartesian[nodeIdx, 0];
+				deformation[1, nodeIdx] = shapeGradientsCartesian[nodeIdx, 1];
+				deformation[2, nodeIdx] = shapeGradientsCartesian[nodeIdx, 2];
+			}
+			return deformation;
+		}
+
+		public Matrix BuildShapeFunctionMatrix(double[] shapeFunctions)
+		{
+			return Matrix.CreateFromArray(shapeFunctions, 1, shapeFunctions.Length);
+		}
+
+		public IReadOnlyList<IReadOnlyList<IDofType>> GetElementDofTypes() => dofTypes;
+
+		public void ResetConstitutiveLawModified()
+		{
+			throw new NotImplementedException();
+		}
+
+		public Tuple<double[], double[]> CalculateResponse(double[] localDisplacements)
+		{
+			throw new NotImplementedException();
+		}
+
+		public double[] CalculateResponseIntegral()
+		{
+			throw new NotImplementedException();
+		}
+
+		public double[] CalculateResponseIntegralForLogging(double[] localDisplacements)
+		{
+			throw new NotImplementedException();
```

Comment:
Method should be implemented as with CalculateResponseIntegral but with logging (check Constitutive)